h2/stream.py

```diff
@@ -633,19 +700,33 @@ def send_alt_svc(self, previous_state):
     (StreamState.CLOSED, StreamInputs.RECV_ALTERNATIVE_SERVICE):
         (None, StreamState.CLOSED),
 
-    # While closed, all other received frames should cause RST_STREAM
-    # frames to be emitted. END_STREAM is always carried *by* a frame,
-    # so it should do nothing.
+    # An endpoint that receives any frame other than PRIORIT after receiving
+    # a RST_STREAM MUST treat that as a stream error of type STREAM_CLOSED.
+    # An endpoint that receives any frames after receiving a frame with the
+    # END_STREAM flag set MUST treat that as a connection error of type
+    # STREAM_CLOSED.
     (StreamState.CLOSED, StreamInputs.RECV_HEADERS):
-        (H2StreamStateMachine.send_reset, StreamState.CLOSED),
+        (H2StreamStateMachine.recv_on_closed_stream, StreamState.CLOSED),
     (StreamState.CLOSED, StreamInputs.RECV_DATA):
-        (H2StreamStateMachine.send_reset, StreamState.CLOSED),
-    (StreamState.CLOSED, StreamInputs.RECV_PUSH_PROMISE):
-        (H2StreamStateMachine.push_on_closed_stream, StreamState.CLOSED),
+        (H2StreamStateMachine.recv_on_closed_stream, StreamState.CLOSED),
     (StreamState.CLOSED, StreamInputs.RECV_END_STREAM):
-        (None, StreamState.CLOSED),
+        (H2StreamStateMachine.recv_on_closed_stream, StreamState.CLOSED),
```

Comment:
This should probably stay as `None`. Note the previous comment about `END_STREAM`.